lib/src/isolate_html.dart

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+Future<Uri> resolveUri(Uri uri) {
+  if (uri.scheme == "package") {
+    uri = Uri.parse("packages/${uri.path}");
```

Comment:
Check that the input is a *valid* package URI, or at least:
```dart
Future<Uri> resolveUri(Uri uri) {
  if (uri.hasScheme("package")) {
    var path = uri.path;
    if (uri.hasAuthority || uri.hasQuery || uri.hasFragment || 
        path.startsWith("/") || path.startsWith("./") || path.startsWith("../")) {
      throw ArgumentError.value(uri, "uri", "Not a valid package URI");
    }
    return Future<Uri>.value(Uri.base.resolve("packages/$path"));
  }
  return Future<Uri>.value(Uri.base.resolve(uri));
}
```



*If* this is the correct resolution (I have no idea what web deployment does, and whether it's in any way consistent), then I'd prefer: